utils/unittest/unittest.go

```diff
@@ -394,19 +394,31 @@ func NetworkTopology() network.Topology {
 
 // CrashTest safely tests functions that crash (as the expected behavior) by checking that running the function creates an error and
 // an expected error message.
-func CrashTest(t *testing.T, scenario func(*testing.T), expectedErrorMsg string, testName string) {
+func CrashTest(
+	t *testing.T,
+	scenario func(*testing.T),
+	expectedErrorMsg string,
+	expectedStatus ...int,
+) {
 	if os.Getenv("CRASH_TEST") == "1" {
 		scenario(t)
 		return
 	}
 
-	cmd := exec.Command(os.Args[0], "-test.run="+testName)
+	cmd := exec.Command(os.Args[0], "-test.run="+t.Name())
 	cmd.Env = append(os.Environ(), "CRASH_TEST=1")
 
 	outBytes, err := cmd.Output()
 	// expect error from run
 	require.Error(t, err)
-	require.Contains(t, "exit status 1", err.Error())
+
+	if len(expectedStatus) == 0 {
```

Comment:
I feel it's too lenient to allow expecting ANY non-zero status. The function under test could crash for some other reason and the unit test could pass, giving us a false sense of security.

Reply:
that's a fair comment.  in that case, maybe provide a `CrashTestWithExpectedStatus` method that requires non-empty list, and convert `CrashTest` into a wrapper calling `CrashTestWithExpectedStatus` with `[]int{1}`

Reply:
Why complicate it with a wrapper when we can just change `CrashTest` to take a single `int` of the expected exit code:

```
func CrashTest(
	t *testing.T,
	scenario func(*testing.T),
	expectedErrorMsg string,
	expectedExitCode int,
)
```

I don't see a case where we would want to pass in more than a single expected exit code.